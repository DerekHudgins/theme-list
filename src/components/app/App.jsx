import React from 'react';
import CharacterList from '../characters/CharacterList';
import Header from '../header/Header';
import { useTheme } from '../context/ThemeProvider';

export default function App() {
  const theme = useTheme();
  return (
    <div
      style={
        theme
          ? { backgroundColor: 'white', color: 'black' }
          : { backgroundColor: 'black', color: 'white' }
      }
    >
      <Header />
      <CharacterList />
    </div>
  );
}
