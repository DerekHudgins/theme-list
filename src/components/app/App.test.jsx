import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';
import ThemeProvider from '../context/ThemeProvider';

describe('App component', () => {
  it('renders App', async () => {
    const { container } = render(
      <ThemeProvider>
        <App />
      </ThemeProvider>
    );

    const ul = await screen.findByRole('character-list', { name: 'Character' });
    expect(ul).not.toBeEmptyDOMElement;
    expect(container).toMatchSnapshot();
  });

  it('changes the toggle for light/dark', async () => {
    render(
      <ThemeProvider>
        <App />
      </ThemeProvider>
    );
    const button = screen.getByRole('button');

    fireEvent.change(button, { target: { value: 'black' } });
    expect(button.value).toBe('black');
  });
});
