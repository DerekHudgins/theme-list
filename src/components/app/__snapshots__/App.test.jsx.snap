// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<div>
  <div
    style="background-color: white; color: black;"
  >
    <header
      style="background-color: white; color: black;"
    >
      <button
        role="button"
      >
        changeTheme
      </button>
    </header>
    <ul
      aria-label="Character"
      role="character-list"
    />
  </div>
</div>
`;
